File: ConstructionOPS/constructionops.client/src/feautures/projects/Contracts/CreateProjectRequest.ts
// contracts/CreateProjectRequest.ts

export interface CreateProjectRequest {
    companyId: string;
    clientId: string;
    projectCode: string;
    name: string;
    description?: string;
    startDate?: string;
    plannedEndDate?: string;
    contractValue: number;
    currencyCode: string;
}

File: ConstructionOPS/ConstructionOPS.Server/Feautures/Projects/Contracts/CreateProjectRequest.cs
namespace ConstructionOPS.Server.Features.Projects.Contracts;

public sealed record CreateProjectRequest
(
    Guid CompanyId,
    Guid ClientId,

    string ProjectCode,
    string Name,

    string? Description,

    DateOnly? StartDate,
    DateOnly? PlannedEndDate,

    decimal ContractValue,

    string CurrencyCode
);